pandas/core/strings.py

```diff
@@ -323,15 +328,24 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     n : int, default -1 (all)
         Number of replacements to make from start
-    case : boolean, default True
-        If True, case sensitive
+    case : boolean, default None
```

Comment:
I like the idea, especially because of the `object` is `True` property. This would make it more portable to other search-pattern functions.

The code is much simpler:
```python
    if not case:
        flags |= re.IGNORECASE

    is_compiled_re = is_re(pat)
    if is_compiled_re and flags:
        raise ValueError("case and flags must be default values"
                         " when pat is a compiled regex")

    use_re = is_compiled_re or len(pat) > 1 or flags or callable(repl)
```

Here's the documentation I'd imagine works well with that. @jreback, can you please let me know what you think before I run the whole thing through again with the other changes above?

```python
    case : boolean, default True (case sensitive)
        Must be True if `pat` is a compiled regex
```